openhands/runtime/remote/runtime.py

```diff
@@ -254,14 +248,50 @@ def _parse_runtime_response(self, response: requests.Response):
                 {'X-Session-API-Key': start_response['session_api_key']}
             )
 
-    @retry(
-        stop=stop_after_attempt(60) | stop_if_should_exit(),
-        wait=wait_fixed(2),
-        retry=retry_if_exception_type(RuntimeError),
-        reraise=True,
-    )
     def _wait_until_alive(self):
         logger.info(f'Waiting for runtime to be alive at url: {self.runtime_url}')
+        # send GET request to /runtime/<id>
+        pod_running = False
+        max_not_found_count = 12  # 2 minutes
+        not_found_count = 0
+        while not pod_running:
+            runtime_info_response = send_request_with_retry(
+                self.session,
+                'GET',
+                f'{self.config.sandbox.remote_runtime_api_url}/runtime/{self.runtime_id}',
+                timeout=10,
```

Comment:
```suggestion
                timeout=5,
```

10s seems way too long